examples/stable_diffusion_xl/gm/util/util.py

```diff
@@ -220,3 +221,26 @@ def auto_mixed_precision(network, amp_level="O0"):
     else:
         raise ValueError("The amp level {} is not supported".format(amp_level))
     return network
+
+
+class LossMonitor(ms.Callback):
+    def __init__(self):
+        self.s = time.time()
+        self.last_step = -1
+
+    def on_train_step_begin(self, run_context):
+        self.s = time.time()
+        cb_params = run_context.original_args()
+        self.last_step = cb_params.cur_step_num
+
+    def on_train_step_end(self, run_context):
+        t = time.time()
+
+        cb_params = run_context.original_args()
+        cur_epoch_num = cb_params.get("cur_epoch_num", 1)
+        cur_step_in_epoch = (cb_params.cur_step_num - 1) % cb_params.batch_num + 1
+        loss = cb_params.net_outputs[0]
+        print(
+            f"epoch: {cur_epoch_num} step: {cur_step_in_epoch}, loss is {loss}, average step time {(t-self.s)*10} ms.",
```

Comment:
Use logging instead of printing. And in data sink mode, the calculated time will be incorrect as it goes through multiple steps in each run.

Reply:
Now the whold SDXL project uses print. It will be modified when the logging module is added.